hierarchicalforecast/core.py

```diff
@@ -90,170 +102,314 @@ class HierarchicalReconciliation:
-            if level_outside_domain and (intervals_method in ['normality', 'permbu']):
-                raise Exception('Level outside domain, send `level` list in [0,100)')
+        if level is not None:
+            level_outside_domain = np.any(
+                (np.array(level) < 0) | (np.array(level) >= 100)
+            )
+            if level_outside_domain and (intervals_method in ["normality", "permbu"]):
+                raise ValueError(
+                    "Level must be a list containing floating values in the interval [0, 100)."
+                )
 
         # Declare output names
-        drop_cols = ['ds', 'y'] if 'y' in Y_hat_df.columns else ['ds']
-        model_names = Y_hat_df.drop(columns=drop_cols, axis=1).columns.to_list()
+        model_names = list(
+            set(Y_hat_df_nw.columns) - set([id_col, time_col, target_col])
+        )
 
         # Ensure numeric columns
-        if not len(Y_hat_df[model_names].select_dtypes(include='number').columns) == len(Y_hat_df[model_names].columns):
-            raise Exception('`Y_hat_df`s columns contain non numeric types')
-            
-        #Ensure no null values
-        if Y_hat_df[model_names].isnull().values.any():
-            raise Exception('`Y_hat_df` contains null values')
-        
-        pi_model_names = [name for name in model_names if ('-lo' in name or '-hi' in name or '-median' in name)]
-        model_names = [name for name in model_names if name not in pi_model_names]
-        
-        # TODO: Complete y_hat_insample protection
-        if intervals_method in ['bootstrap', 'permbu'] and Y_df is not None:
-            if not (set(model_names) <= set(Y_df.columns)):
-                raise Exception('Check `Y_hat_df`s models are included in `Y_df` columns')
+        for model in model_names:
+            if Y_hat_df_nw.schema[model] not in [
+                nw.Int8,
+                nw.Int16,
+                nw.Int32,
+                nw.Int64,
+                nw.UInt8,
+                nw.UInt16,
+                nw.UInt32,
+                nw.UInt64,
+                nw.Float32,
+                nw.Float64,
+            ]:
+                raise Exception(
+                    f"Column `{model}` in `Y_hat_df` contains non-numeric values. Make sure no column in `Y_hat_df` contains non-numeric values."
+                )
+            if Y_hat_df_nw[model].is_null().any():
+                raise Exception(
+                    f"Column `{model}` in `Y_hat_df` contains null values. Make sure no column in `Y_hat_df` contains null values."
+                )
 
-        uids = Y_hat_df.index.unique()
+        # TODO: Complete y_hat_insample protection
+        model_names = [
+            name
+            for name in model_names
+            if not ("-lo" in name or "-hi" in name or "-median" in name)
+        ]
+        if intervals_method in ["bootstrap", "permbu"] and Y_df_nw is not None:
+            if not (set(model_names) <= set(Y_df_nw.columns)):
```

Comment:
Would be useful to save this in order to show the missing columns in the error message.

Reply:
I don't understand what you mean?